sxt/base/field/element.h

```diff
@@ -0,0 +1,29 @@
+/** Proofs GPU - Space and Time's cryptographic proof algorithms on the CPU and GPU.
+ *
+ * Copyright 2023-present Space and Time Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <concepts>
+
+namespace sxt::basfld {
+//--------------------------------------------------------------------------------------------------
+// element
+//--------------------------------------------------------------------------------------------------
+template <class T>
+concept element = requires(T& elm) {
+  { elm.num_limbs_v } noexcept -> std::convertible_to<size_t>;
```

Comment:
This doesn't look like it's enough to specify a functional concept.

Could you please outline how you intend to use it? If the only requirement is specifying num_limbs_v I wouldn't think a concept is necessary.

Reply:
You're right. I was planning on adding to the concept as I refactored, but an outline is a good idea to determine if I am going in the right direction. To support MSM with BN254 and bls12-381, I will need the `num_lims_v` and access to the modulus. I think post MSM this concept's requirements could grow to encompass the values defined in Arkworks field definition ([example](https://github.com/arkworks-rs/curves/blob/master/bls12_381/src/fields/fq.rs)).

I created a [PR](https://github.com/spaceandtimelabs/blitzar/pull/59) that has the refactoring of the `field12/operation/add` component which uses the concept. Maybe it can help show you how I was planning on using the concept and if it is best direction?